### ISSUE TITLE
After upgrading to ReactJS.Net 3 we faced a memory leak

### ISSUE THREAD

**Issue by csoulioti** [reactjs/React.NET#343]
I would like to report one problem that came up after upgrading to ReactJS.Net 3. We have an azure cloud service, and our application, which is an mvc application, performs heavy server side rendering. We have been having a problem where 1 or more of our Azure instances would become inaccessible. We noticed that the RAM usage of w3wp every hour is increasing. We set the SetReuseJavaScriptEngines flag to false and in this case we do not have memory leak but the cpu usage is over the limits. We reverted back to previous version and this is not happening. Does anyone has any idea of what it may cause this memory leak?

**Comment by Daniel15** [comment]
Interesting... I wonder if this has to do with the JavaScriptEngineSwitcher upgrade (I upgraded it to version 2.0 in ReactJS.NET 3.0). So the memory usage doesn't go up in the same way with older versions of ReactJS.NET?

**Comment by csoulioti** [comment]
Yes exactly, with older versions of ReactJS.NET, the memory usage doesn't go up in this way. Do you have any idea of how could I test it somehow to see if the JavaScriptEngineSwitcher is the reason of?

**Comment by markau** [comment]
I am experiencing this issue too. I note that ReactJS.NET 3.0 installed the engine switching libraries at 2.0.3 but they are now up to 2.2.0. I can only hope that helps otherwise I have wasted a lot of effort.

@csoulioti Did you find a solution? 

Is ReactJS.NET up to a production environment?

**Comment by gunnim** [comment]
We have the same issue, setReuseJavascriptengines false fixes.

.SetAllowMsieEngine(false) 
+
and install javascriptEngineSwitcher.v8.native.win-x64 
also fixes.
Leading me to believe that React .net is trying to use the msie engine and that engine seems to be buggy when it comes to disposing.
I recreate the memory leak by touching the web.config and sending a web request. This usually fires up a new app domain and grows the memory. If i do this 5 times memory can grow to 2gb and stays that way if i check back an hour later and 4-6 .net app domains (checked with procexp from sysInternals).
Some dispose issue @Daniel15  ?

**Comment by gunnim** [comment]
Issue is not present with the 2.4.10 version of the JavascriptEngineSwitcher libraries (despite not having the v8.native.win-x64 package)
